dl/src/main/scala/com/intel/analytics/bigdl/optim/SGD.scala

```diff
@@ -32,17 +32,19 @@ class SGD[@specialized(Float, Double) T: ClassTag](implicit ev: TensorNumeric[T]
     config: Table, state: Table = null): (Tensor[T], Array[T]) = {
 
     val _state = if (state == null) config else state
-    val lrSchedule = config.get[LearningRateSchedule]("learningRateSchedule").getOrElse(Default())
-    lrSchedule.updateHyperParameter(config, _state)
-
-    val wd = config.get[Double]("weightDecay").getOrElse(0.0)
-    val mom = config.get[Double]("momentum").getOrElse(0.0)
-    val damp = config.get[Double]("dampening").getOrElse(mom)
-    val nesterov = config.get[Boolean]("nesterov").getOrElse(false)
-    val lrs = config.get[Tensor[T]]("learningRates").getOrElse(null)
-    val wds = config.get[Tensor[T]]("weightDecays").getOrElse(null)
-
-    require(!nesterov || (mom > 0 && damp == 0),
+    val scheduler = config.getOrElse[HyperParameterScheduler]("hyperParameterScheduler", Default())
+    scheduler.updateHyperParameter(config, _state)
+
```

Comment:
We should not give two interface(e.g. wd and weightDecay for weightDecay), or the code is confused and easy to make bug.

If you touch other hyperparameter instead of only learningRate, you should make the interface align with learningRate, like a clr and learningRate, cweightDecay and weightDecay, etc.

So SGD only read hyperparameter from clr, cweightDecay... And user specify learningRate and weightDecay in their configuration